lib/default/default/common.lua

```diff
@@ -0,0 +1,42 @@
+local awful = require("awful")
+local beautiful = require("beautiful")
+local hotkeys_popup = require("awful.hotkeys_popup")
+local common = {}
+
+
+function common.init(args)
+   local args = args or {}
+   -- {{{ Variable definitions
+
+   -- This is used later as the default terminal and editor to run.
+   common.terminal = args.terminal or "xterm"
```

Comment:
idk what is the faith of this PR as a whole, however if the idea itself will be approved in this way, i suggest using smth like `common._data.terminal` or so instead of storing variables just on the first module level of that lib

Reply:
noted
btw, what do you think of the idea itself?

Reply:
i'm using a similar layout in my config, but with 2 modules instead of 1, as it's a bit bigger than default one

but at the same time idk how would it apply for the most of users in terms of ease of changing some small value in default config (as it will require making a copy of one of default-submodules into local dir for doing that change)

Reply:
the fact that you and other hardcore users use their own implementations of big frameworks on top of awesome is the whole reason why some version of this should be merged. And ideally a culture of easily sharable code should be established.

as to the most users and copying, they already do that, they copy one file and change 5 lines in it, and than they add 5 more lines, and more and more, and it becomes bloated crap, all because they started with monolithic huge file.  With this approach they will have to copy and edit only files that they need, and for adding new widgets and new key bindings they don't need to edit any default configs)

Reply:
i really don't think my config would be any smaller, as i have almost any line of default config redefined, including default hotkeys, layouts, signals, panels, whatever -- so actually on this question it might be more valuable to get the feedback from people who have their config 50+ % match the default one

Reply:
Hi, I'm rather new to awesome wm, and haven't changed large chunks of the code.

I think that most new comers start by changing the wibar, theme and client key bindings (shortcuts to launch programs) sections. Also,  dividing the config to multiple files can help users to maintain the same pattern when creating their own versions of the changed files. I myself struggle with deciding how to divide the monolithic config file into smaller, more easily editable files.